src/backend/executor/nodeDML.c

```diff
@@ -348,9 +445,30 @@ ExecEndDML(DMLState *node)
 	ReleaseTupleDesc(node->junkfilter->jf_cleanTupType);
 
 	ExecFreeExprContext(&node->ps);
-	ExecClearTuple(node->ps.ps_ResultTupleSlot);
+	if (node->ps.ps_ResultTupleSlot != NULL)
+		ExecClearTuple(node->ps.ps_ResultTupleSlot);
 	ExecClearTuple(node->cleanedUpSlot);
 	ExecEndNode(outerPlanState(node));
 	EndPlanStateGpmonPkt(&node->ps);
 }
+
+void
+ExecSquelchDML(DMLState *node)
+{
+	/*
+	 * DML nodes must run to completion when asked to Squelch so
+	 * that we don't risk losing modifications which should be performed
+	 * regardless of any LIMIT's or other forms for projections which could
+	 * end up causing a squelch to happen.
+	 */
+	for (;;)
+	{
+		TupleTableSlot *result;
+
+		result = ExecDML(node);
+		if (!result)
+			break;
+	}
```

Comment:
We don't need the return value of  ExecDML, we should only check if the function returns NULL. What about
```suggestion
	while (ExecDML(node) != NULL);
```
to make the code simpler?

Reply:
done